### PR TITLE
EWPP-384: Add margin to blocks if they are part of the content region.

### PR DESCRIPTION
**Pull request by imanoleguskiza**
### Description

Add margin to blocks if they are part of the content region.

### Change log

- Added: Add margin to blocks if they are part of the content region.
- Changed:
- Deprecated:
- Removed:
- Fixed:
- Security:

### Commands

```sh
[Insert commands here]

```

### CODE REVIEW

File: oe_theme.theme [hunk 1]
```diff
@@ -1632,3 +1631,14 @@ function oe_theme_preprocess_field__media__oe_media_iframe__video_iframe__oe_the
...
+
+/**
+ * Implements hook_theme_suggestions_HOOK_alter().
+ *
+ * Add block region as a suggestion.
+ */
+function oe_theme_theme_suggestions_block_alter(array &$suggestions, array $variables) {
+  /** @var \Drupal\block\Entity\Block $block */
+  $block = \Drupal::entityTypeManager()->getStorage('block')->load($variables['elements']['#id']);
+  $suggestions[] = 'block__' . $block->getRegion();
```
Comment (ademarco) [inline oe_theme.theme]: Let's prefix this with `region_` like 
```
   $suggestions[] = 'block__region_' . $block->getRegion();
```

File: oe_theme.theme [hunk 2]
```diff
@@ -1632,3 +1631,14 @@ function oe_theme_preprocess_field__media__oe_media_iframe__video_iframe__oe_the
...
 }
+
+/**
+ * Implements hook_theme_suggestions_HOOK_alter().
+ *
+ * Add block region as a suggestion.
+ */
+function oe_theme_theme_suggestions_block_alter(array &$suggestions, array $variables) {
+  /** @var \Drupal\block\Entity\Block $block */
+  $block = \Drupal::entityTypeManager()->getStorage('block')->load($variables['elements']['#id']);
```
Comment (ademarco) [inline oe_theme.theme:1642]: Don't we have the block object in $variables already?

File: templates/blocks/block--content.html.twig [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+#}
+<div{{ attributes.addClass('ecl-u-mt-l').addClass('ecl-u-mb-l') }}>
+  {{ title_prefix }}
+  {% if label %}
+    <h2{{ title_attributes }}>{{ label }}</h2>
+  {% endif %}
+  {{ title_suffix }}
+  {% block content %}
+    {{ content }}
+  {% endblock %}
```
Comment (ademarco) [inline templates/blocks/block--content.html.twig]: You can extend and use the {{ parent() }} twig directive. Look for parent() in our templates, we did that somewhere else.

**Review by ademarco — CHANGES_REQUESTED** [review]

File: oe_theme.theme (hunk 2 above)
Comment (imanoleguskiza) [reply oe_theme.theme:1642]: Sadly no, the configuration of the block is there but the region information is nowhere to be found.

**Review by nagyad — APPROVED** [review]

**Review by ademarco — APPROVED** [review]

**Pull request merged**